### ISSUE TITLE
Investigate and fix Hungary

### ISSUE THREAD

**Issue by IndyHurt** [whosonfirst/whosonfirst-data#28]
WOF has 19 counties and the capital city.

For reference, Statoids lists 19 counties, 1 capital city, 23 urban counties / county boroughs/ cities of county right.  

WOF is missing the 23 urban counties / county boroughs / cities of county right.

**Comment by stepps00** [comment]
After investigating, it looks like Who's On First does have all of the administrative records in Hungary.

The 23 urban counties / cities of right are actually in Who's On First as `locality` records, which is correct. The cities of right are parented by the county records in Hungary - take the example of [Hódmezővásárhely](https://whosonfirst.mapzen.com/spelunker/id/101790425/#13/46.4204/20.3358). As listed on [the wikipedia page](https://en.wikipedia.org/wiki/Csongr%C3%A1d_County), this city of right is administratively parented by Csongr County. It would be incorrect to change the placetypes of any of these features in Who's On First. 

Related: [This issue](https://github.com/whosonfirst-data/whosonfirst-data/issues/250) is still open, but once the PR is merged for that issue, WOF will have the correct parenting for all macroregions in Hungary.

Suggest ensuring that the hierarchies of administrative places are correct after merging [this PR](https://github.com/whosonfirst-data/whosonfirst-data/pull/304), then closing this issue.

**Comment by nvkelso** [comment]
Those 23 urban counties / cities should get a placetype_alt of `region`.

**Comment by nvkelso** [comment]
Looks like our handful of `macroregion` features should get mz:hierarchy_label = 0 and be marked as wof:statistical_gore = 1.

**Comment by ImreSamu** [comment]
the new hierarchical system  **2018.Jan.01** - ( related #1209 ) 


EU  hierarchical system  | whosonfirst placetype     |  hungary - statistics 
-----| ----------   | --------------------
NUTS0 | country  | Hungary : https://spelunker.whosonfirst.org/id/85633237/ 
NUTS1 | macroregion |  3 regions at NUTS 1  https://www.ksh.hu/docs/teruletiatlasz/nuts1.pdf
NUTS2 | region |   8 regions at NUTS 2  https://www.ksh.hu/docs/teruletiatlasz/regions.pdf
NUTS3 | ~county |  20 regions at NUTS 3 level.
LAU ( Local Administrative Units)  | locality |    3155   (  #918 )



### imho this is a NUTS1  ~ macroregion  ( count=3)

https://www.ksh.hu/docs/teruletiatlasz/nuts1.pdf

```
404227489 Central Hungary — this is a macroregion in Hungary  
Transdanubia = ( 404227491 Southern Transdanubia  + 404227481 Western Transdanubia  + 404227479 Central Transdanubia  )
Great Plain and North = ( 404227487 Southern Great Plain + 404227483 Northern Great Plain + 404227477 Northern Hungary )
```

### imho this is a NUTS2 ~ region   ( count=8)

https://www.ksh.hu/docs/teruletiatlasz/regions.pdf

```
404227487 Southern Great Plain — this is a macroregion->region in Hungary
404227483 Northern Great Plain — this is a macroregion->region in Hungary
404227491 Southern Transdanubia — this is a macroregion->region in Hungary
404227477 Northern Hungary — this is a macroregion->region in Hungary
404227481 Western Transdanubia — this is a macroregion->region in Hungary
404227479 Central Transdanubia — this is a macroregion->region in Hungary
404227489 Central Hungary — this is a macroregion in Hungary  ---> Split [ Budapest, Pest ] )
```

###  imho this is a  NUTS3 ~ "county"   ( count=20)
```
85684839 Baranya — this is a region->county in Hungary  ( https://en.wikipedia.org/wiki/Baranya_County )
85684841 Tolna — this is a region->county in Hungary
85684845 Somogy — this is a region->county in Hungary
85684849 Zala — this is a region->county in Hungary
85684855 Csongrád — this is a region->county in Hungary
85684859 Bács-Kiskun — this is a region->county in Hungary
85684863 Vas — this is a region->county in Hungary
85684871 Veszprém — this is a region->county in Hungary
85684875 Békés — this is a region->county in Hungary
85684879 Fejér — this is a region->county in Hungary
85684881 Budapest — this is a region->county in Hungary
85684885 Komárom-Esztergom — this is a region->county in Hungary
85684891 Jász-Nagykun-Szolnok — this is a region->county in Hungary
85684895 Gyor-Moson-Sopron — this is a region->county in Hungary
85684899 Pest — this is a region->county in Hungary
85684903 Hajdú-Bihar — this is a region->county in Hungary
85684909 Heves — this is a region->county in Hungary
85684913 Nógrád — this is a region->county in Hungary
85684915 Szabolcs-Szatmár-Bereg — this is a region->county in Hungary
85684919 Borsod-Abaúj-Zemplén — this is a region->county in Hungary
```

**Comment by stepps00** [comment]
Thanks for the notes, @ImreSamu - finally getting to take a look at Hungary again. It looks like there are a few issues with the way WOF currently classifies Hungary admin records.

Potential issues:
- WOF is missing the [NUTS1](https://www.ksh.hu/docs/teruletiatlasz/nuts1.pdf) divisions. Since WOF is missing the NUTS1 divisions, if we do decide to import this division, all placetypes may need to be adjusted (macroregion records moved to region, region records moved to county) to make room for these new macroregions.

- WOF is missing [174 districts](https://en.wikipedia.org/wiki/Districts_of_Hungary) of Hungary

- WOF is missing complete LAU coverage (towns, towns with county rights, large municipalities, and municipalities)

In short, I agree with your assessment above. Since it would be a "clean" import (we don't have this placetype in WOF), I'd like to import the [districts](https://en.wikipedia.org/wiki/Districts_of_Hungary) at the localadmin placetype first, then work on cleaning up the existing placetypes.

If you have a source for data at the district level, feel free to pass that along.

**Comment by stepps00** [comment]
https://geoshop.hu/en/products/mkh

Various sites have directed me to this portal, but the data is not open/can't be imported into WOF. I'll append to this comment as I find new potential sources.

**Comment by justinelliotmeyers** [comment]
@nvkelso @stepps00 test this out: http://geonetwork.inspire.fomi.hu/geonetwork/srv/eng/catalog.search#/metadata/618c85bf-f6f4-49b9-85d3-88ffd7f6d4da

Enjoy ;)

**Comment by stepps00** [comment]
See also: http://inspire.fomi.hu/inspire/atom/data/AdministrativeUnits_hu_v1.1.zip

Which seems to be populated from: http://inspire.gov.hu/geonetwork/srv/eng/catalog.search#/metadata/b9c3b9ca-75b0-4e06-b519-d0a85d4fe4ab

Licensing terms unclear.

**Comment by justinelliotmeyers** [comment]
@stepps00 thanks for cleaning up that dup!!

**Comment by nvkelso** [comment]
@ImreSamu 👋 does Hungary have a modern web portal for downloading administrative boundaries (polygons) in shapefile or GeoPackage format?

**Comment by justinelliotmeyers** [comment]
might be in here? https://map.mbfsz.gov.hu/

**Comment by justinelliotmeyers** [comment]
<img width="812" alt="hu" src="https://user-images.githubusercontent.com/5324316/233811993-ace545f4-bd9d-4888-aa8c-03ee2a07c965.png">
think these are the critters you are seeking

**Comment by nvkelso** [comment]
[nvkelso](https://github.com/nvkelso) commented [3 days ago](https://github.com/whosonfirst-data/whosonfirst-data/issues/2076#issue-1676047700) in #2076:

Hungary has 3155 Települések (Settlements) and they are used as NUTS LAU and could be imported into WOF as `localadmin`.

See also:

- https://en.wikipedia.org/wiki/NUTS_statistical_regions_of_Hungary
- https://en.wikipedia.org/wiki/List_of_cities_and_towns_of_Hungary

![image](https://user-images.githubusercontent.com/853051/233271573-b5901686-c9e3-483c-9f11-aa7ed1baff2d.png)

Let's keep this issue open and close that one.

**Comment by justinelliotmeyers** [comment]
![image](https://user-images.githubusercontent.com/5324316/233812111-1096744b-792c-4e20-95bb-0e606d5cf103.png)

Lets Go

**Comment by nvkelso** [comment]
Bingo! 

Last catch is what is the license? 

The scale also looks to be 1:250,000 instead of 1:20,000 – so great coverage but so-so precision compared with rest of EU countries.

> **SITE TERMS OF USE:** The maps and figures on the map portal of the MBFSZ (map.mbfsz.gov.hu), as well as the textual references, information, and legends related to them, are the intellectual property of the MBFSZ. For any kind of official use, contact the MBFSZ in advance!

**Comment by ImreSamu** [comment]
@nvkelso :
> wave does Hungary have a modern web portal 
> for downloading administrative boundaries (polygons) in shapefile or GeoPackage format?

with an open License?   
I don't know about it, and I don't have high hopes, but I will try to find out.

imho: 
Instead of a Hungarian government portal, I would look for an EU-based geo portal where this data is available. 

For example:   **EU - LOCAL ADMINISTRATIVE UNITS (LAU)**
https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units/lau#lau19
this one here seems to have local-level data for the whole EU. At a glance, the data for Hungary also looks good.
the only problems the [license restrictions](https://ec.europa.eu/eurostat/web/gisco/geodata/reference-data/administrative-units-statistical-units) : 
*  _"the data will not be used for commercial purposes;"_

LAU 2021 ~ Zooming in on Hungary : 
![image](https://user-images.githubusercontent.com/2217320/233819772-8f8728f1-d102-4ee6-bd79-4589ca89d0cd.png)

LAU 2021 - Full coverage:
![image](https://user-images.githubusercontent.com/2217320/233819839-e8195242-3b5c-4e74-9518-f81b33595c98.png)

**Comment by justinelliotmeyers** [comment]
June 2024 the Eurostat stuff will be cc-by I was told.

**Comment by nvkelso** [comment]
@justinelliotmeyers Thoughts? Can you check if the Eurostats stuff did go CC-BY in June 2024?

**Comment by justinelliotmeyers** [comment]
https://ec.europa.eu/eurostat/web/main/help/copyright-notice

Also asking my colleague there. He will give me a better idea too.

@nvkelso thoughts about this: https://www.ksh.hu/copyright

has two licenses... I defer to no, but maybe you disagree?

**Comment by ImreSamu** [comment]
I understand it as follows: 
* the numbers, statistics,   ✅   [  default CC BY 4.0    ~ population , xls tables  ]
* AND GISCO/Geodata (SHP)   ⛔  ( _"There are specific [download rules](https://ec.europa.eu/eurostat/web/gisco/geodata/statistical-units) for the datasets shown "_ --> ""the data will not be used for commercial purposes"" )

have different licenses.




### Geodata (SHP)  - "Local administrative units (LAU)"    - "the data will not be used for commercial purposes"  ⛔ 

https://ec.europa.eu/eurostat/web/gisco/geodata/statistical-units/local-administrative-units

![Image](https://github.com/user-attachments/assets/a16d48e0-1978-4a1c-8913-0f579b5154e9)

--> "There are specific [download rules](https://ec.europa.eu/eurostat/web/gisco/geodata/statistical-units) for the datasets shown "

![Image](https://github.com/user-attachments/assets/59a15db8-4094-4b63-b154-e58a7e32d489) 

--> "the data will not be used for commercial purposes"


### Metadata : Local administrative units (LAU)  ( XLS )  -  CC BY 4.0  -  ✅ 

https://ec.europa.eu/eurostat/web/nuts/local-administrative-units 
--> LAU – NUTS 2024, EU-27 and EFTA / available candidate countries	Partially validated ;  2024 :   XLS 

![Image](https://github.com/user-attachments/assets/e6f9047a-a067-4e15-800e-3a4f4d2d79aa)

And Merging with Wikidata info (  ~ Coordinate location  )  - we have a simple POINT info

run this SPARQL query in  https://query.wikidata.org/
```SPARQL
SELECT 
  ?item                                                # Wikidata ID
  (SAMPLE(?label_hu) AS ?hungarian_label)               # Hungarian label
  (GROUP_CONCAT(DISTINCT ?ksh; separator=";") AS ?kshConcat)                      # KSH code(s)
  (GROUP_CONCAT(DISTINCT ?whosOnFirst; separator=";") AS ?whosOnFirstConcat)      # Who's on First ID (P6766)
  (GROUP_CONCAT(DISTINCT ?geoNames; separator=";") AS ?geoNamesConcat)            # GeoNames ID (P1566)
  (GROUP_CONCAT(DISTINCT ?website; separator=";") AS ?websiteConcat)              # Official website (P856)
  (GROUP_CONCAT(DISTINCT ?postalCode; separator=";") AS ?postalCodeConcat)        # Postal code (P281)
  (GROUP_CONCAT(DISTINCT STR(?coord); separator=";") AS ?coordConcat)             # Coordinate location (P625)
  (GROUP_CONCAT(DISTINCT ?country; separator=";") AS ?countryConcat)              # Country (P17)
  (GROUP_CONCAT(DISTINCT ?admEntity; separator=";") AS ?admEntityConcat)          # Administrative territorial entity (P131)
  (GROUP_CONCAT(DISTINCT ?unLocode; separator=";") AS ?unLocodeConcat)            # UN/LOCODE (P1937)
  (GROUP_CONCAT(DISTINCT ?woeid; separator=";") AS ?woeidConcat)                  # WOEID (P1281)
WHERE {
  # Filter for items that have a KSH code (P939) and belong to Hungary (P17 wd:Q28)
  ?item wdt:P939 ?ksh.
  ?item wdt:P17 wd:Q28.
  
  # Retrieve the Hungarian label
  OPTIONAL { ?item rdfs:label ?label_hu. FILTER(LANG(?label_hu)="hu") }
  
  # Retrieve additional properties
  OPTIONAL { ?item wdt:P6766 ?whosOnFirst. }
  OPTIONAL { ?item wdt:P1566 ?geoNames. }
  OPTIONAL { ?item wdt:P856 ?website. }
  OPTIONAL { ?item wdt:P281 ?postalCode. }
  OPTIONAL { ?item wdt:P625 ?coord. }
  OPTIONAL { ?item wdt:P17 ?country. }
  OPTIONAL { ?item wdt:P131 ?admEntity. }
  OPTIONAL { ?item wdt:P1937 ?unLocode. }
  OPTIONAL { ?item wdt:P1281 ?woeid. }
}
GROUP BY ?item
ORDER BY ?item
```
Example - outputs: 

![Image](https://github.com/user-attachments/assets/10c00b17-d0b8-4c51-a4bd-5c7b63343438)

RESULTS: 
* [wikidata_2025_02_01_KSH_Local_admins_query.csv](https://github.com/user-attachments/files/18629475/wikidata_2025_02_01_KSH_Local_admins_query.csv)


JOIN HINT: 
* EU LAU code  =  "HU_"+KSH_code (  Hungarian KSH  ) 
   * so LEFT JOIN   EU XLS:   EU LAU  CODE =  `HU_10126`   ==  Wikidata(Győrtelek  kshConcat = `10126` )

**Comment by ImreSamu** [comment]
> thoughts about this: https://www.ksh.hu/copyright

( As a Hungarian - but not as a lawyer. ) 

English version - https://www.ksh.hu/copyright_eng   [ KSH = Hungarian Central Statistical Office (hereinafter: HCSO) ] 
> "HCSO uses standardised international licence [Creative Commons Attribution 4.0 International (CC BY 4.0)](https://creativecommons.org/licenses/by/4.0/), according to which all of the content available on the Website – including tables, figures and infographics – (hereinafter: Content) can be copied, reproduced and redistributed without limitations. Exceptions of this are data files queried from the internal databases of HCSO on specific request, the User is not entitled to use these files for commercial purposes (see point 3.3 in the present Terms of Use)."

I understand this license similarly to the EU STAT license:
* CSV and XLS files are available under the CC BY 4.0 license, but these files do not include geodata.

One interesting aspect is the administrative changes (such as the merging and splitting of municipalities since 1970).  For everything else, I believe the EU LAU data is sufficiently acceptable, even though the population data from KSH/HCSO are more up-to-date.

**Comment by ImreSamu** [comment]
https://www.mapsforeurope.org/   probably  ✅ 
* progress: https://eurogeographics.org/open-maps-for-europe/ome2-progress/



## Open Maps For Europe looks OK:  ( _"the license allows for commercial use"_ )

https://www.mapsforeurope.org/frequently-asked-questions

> Open Maps For Europe 2 (OME2) is a new project that will develop a new production process and technical specification for free-to-use, edge-matched data under a single open licence. Authoritative 1:10 000 scale data for 10 countries will be delivered via the user interface built by the award-winning Open Maps For Europe Project. The project will also enhance the five existing Open Maps For Europe datasets, including the pilot Open Cadastral Map.
> 
> The project started in January 2023 and will conclude in December 2025.
> 
> The OME2 project, co-funded by the European Union, is being delivered by a consortium comprising: EuroGeographics the not-for-profit membership association for European National Mapping, Cadastral and Land Registration Authorities (NMCAs); National Geographic Institute, Belgium; National Institute of Geographic and Forest Information, France; Hellenic Cadastre; General Directorate for the Cadastre, Spain; and Cadastre, Land Registry and Mapping Agency, The Netherlands
> 


> Coverage is confirmed as follows:
> 
> [EuroGlobalMap](https://www.mapsforeurope.org/datasets/euro-global-map) 1:1 million scale open dataset - 55 countries and territories.
> [EuroRegionalMap](https://www.mapsforeurope.org/datasets/euro-regional-map) 1:250,000 scale open dataset – 38 countries and territories from 31 data providers
> [EuroDEM](https://www.mapsforeurope.org/datasets/euro-dem) Digital elevation model – 46 countries and territories from 29 data providers and infilled with SRTM open data
> [Open Gazetteer](https://www.mapsforeurope.org/datasets/open-gazetteer) Pan European Open Gazetteer – 35 countries from 36 data providers
> [Pan-European Imagery](https://www.mapsforeurope.org/datasets/pan-european-imagery) Remote sensing data of the European Union's Earth observation programme Copernicus (Sentinel-2, L1C-L2A) – Full coverage of geographical Europe
> [Open Cadastral Map](https://www.mapsforeurope.org/datasets/cadastral-all) Six agencies contributing data with the aim of increasing coverage.
> [High-Value Large Scale Prototype](https://www.mapsforeurope.org/datasets/hvlsp) 1:10,000 scale open dataset - 5 agencies contributing data across two themes. This will be increased during the OME2 project.
> 

> Which open data licence does Open Maps For Europe use?
> To be able to deliver the Open Maps For Europe datasets, EuroGeographics is using its existing  which members have already agreed to. This licence allows users to distribute, remix, adapt, and build upon the material in any medium or format, so long as attribution is given to the creator. **The license allows for commercial use.** This licence covers EuroGlobalMap, EuroRegionalMap, EuroDEM, the Open Gazetteer and the High Value Large Scale Prototype.

>I downloaded the Open Gazetteer as a dataset but it does not include all the names that are in the search function of the interface which is also using the Open Gazetteer, why is this?
> The Open Gazetteer has been infilled with data from GeoNames, to ensure that the search functionality is as comprehensive as possible for end users when using the interface. We aim to increase the coverage of the Gazetteer as the project progresses.

### Open Gazetteer
https://www.mapsforeurope.org/datasets/open-gazetteer

### Global Maps

![Image](https://github.com/user-attachments/assets/acd228f6-fd9d-46a5-838c-8e17fcbb7cc4)

![Image](https://github.com/user-attachments/assets/d4a3ec85-d3f5-4b77-b75f-2f2bb507dbce)


### Euro -regional -map 

Hungary is missing ..

![Image](https://github.com/user-attachments/assets/41de894e-b3c2-430b-b204-e8cae72aa0b2)

![Image](https://github.com/user-attachments/assets/68d3e9fd-6c7b-4c7f-98c4-6b870a071885)

**Comment by justinelliotmeyers** [comment]
unfortunately GISCO is derived from Eurogeographics products. Which isn't a bad thing, it's just finding Hungary. I'm still snooping around. I know WOF doesn't use OSM, but I think @osm basically imported the ksh settlements, or they are very very close (*i think). I havent looked at the polygons in a few years.

I'll keep looking and see what I find. Hungary is pretty easy. It's just finding the dataset with the proper license.
